app/controllers/events_controller.rb

```diff
@@ -0,0 +1,92 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: [:show, :edit, :destroy, :attend]
+  #before_action :correct_user, only: %i[edit update]
+
+  # GET /events
+  # GET /events.json
+  def index
+    if params[:events].nil? || params[:events][:type] == 'upcoming'
```

Comment:
`[Optional]` I don't think this condition is necessary. You can display both types of events at the same time.